wire-ios-data-model/Tests/Source/Model/Conversation/ZMConversationTests+Messages.swift

```diff
@@ -268,6 +270,8 @@ class ZMConversationMessagesTests: ZMConversationTestsBase {
     }
 
     func testThatLastReadUpdatesInSelfConversationDontExpire() {
+        let backendInfoToken = TemporaryBackendInfoToken(domain: "wire.com")
```

Comment:
@markusfassbender we could use defer here to put everything in same place

idea 2: use a swiftmacro to specify this test will use backend domain X
+ add swiftlint rule to ban usage of BackendInfo in tests and use this instead

Reply:
@samwyndham @netbe thanks for your ideas. The issue is, that one can easily forget to do the reset and then other tests are influenced, e.g. `testThatItReportsToDelegate_WhenFederationHasBeenEnabled`.
So my idea solution was something that does reset automatically..

Reply:
@markusfassbender You could always fix that issue by doing something like the following although we should still clean up for memory reasons.

```swift
BackendInfo.storage = UserDefaults(suiteName:  UUID().uuidString)!
```

This should be safe as in XCTest, each test run uses a different class instance of `BackendInfo`. E.g. setting `BackendInfo.storage` in one test shouldn't affect that property in another test.

Reply:
Hmmm. Maybe I'm wrong about the about that. This is what I had read but just tried it in reality and that is not the behavior.